qiskit_quantuminspire/qi_provider.py

```diff
@@ -0,0 +1,53 @@
+from datetime import datetime, timezone
+from typing import Any, List, Union
+
+from compute_api_client import BackendStatus, BackendType, Metadata
+
+from qiskit_quantuminspire.qi_backend import QIBackend
+
+
+class QIProvider:
+    """List QIBackends integrated with QiskitBackend interface."""
+
+    def __init__(self) -> None:
+        self._qiskit_backends = self._construct_backends()
+
+    def _fetch_qi_backend_metadata(self) -> List[Metadata]:
+        """Fetch backend metadata using api client."""
+        return [Metadata(id=1, backend_id=1, created_on=datetime.now(timezone.utc), data={"nqubits": 6})]
+
+    def _fetch_qi_backend_types(self) -> List[BackendType]:
+        """Fetch backend types from CJM using api client."""
+        backend_type_list = [
+            BackendType(
+                id=2,
+                name="Spin 2",
+                infrastructure="Hetzner",
+                description="Silicon spin quantum computer",
+                image_id="abcd1234",
+                is_hardware=True,
+                features=["multiple_measurements"],
+                default_compiler_config={},
+                native_gateset={"single_qubit_gates": ["X"]},
+                status=BackendStatus.IDLE,
+                default_number_of_shots=1024,
+                max_number_of_shots=2048,
+            ),
+        ]
+        return backend_type_list
+
+    def _construct_backends(self) -> List[QIBackend]:
+        """Construct QIBackend using fetched backendtypes and metadata."""
+        qi_backend_types = self._fetch_qi_backend_types()
+        qi_metadata = self._fetch_qi_backend_metadata()
```

Comment:
We probably shouldn't separately fetch the backend_types and the metadata, as that would make matching the correct metadata with the correct backend a bit annoying. I propose that `_fetch_qi_backend_metadata()` changes to a function that accepts `BackendType` instead of `List[BackendType]`, and is called iteratively while constructing QIBackends.

Reply:
Agreed, changed